lib/core/proxy.c

```diff
@@ -421,6 +440,13 @@ static int on_body(h2o_http1client_t *client, const char *errstr)
     if (self->sending.bytes_inflight == 0)
         do_send(self);
 
+    if (self->client && self->client->sock && overrides && self->client->sock->input->size > overrides->max_buffer_size) {
+        self->await_send = await_send;
+        self->await_send_arg.proxy_generator = self;
+        self->await_send_arg.cb = self->client->sock->_cb.read;
+        h2o_socket_read_stop(self->client->sock);
```

Comment:
How about creating a functions that stops / resumes reading the body in http1client?

I assume that by doing so there would no longer be a need to introduce `await_send_arg_t`, since:
* it is possible to determine if the reader is active by calling `h2o_socket_is_reading(self->client->sock)`
* in http1client the callback function that needs to be passed to `h2o_socket_read_start` is known to be a constant